compiler/rustc_error_codes/src/error_codes/E0508.md

```diff
@@ -39,3 +39,16 @@ fn main() {
     let _value = array[0].clone();
 }
 ```
+
+If you really want to move the value out, you can use a destructuring array
+pattern to move it:
+
+```
+struct NonCopy;
+
+fn main() {
```

Comment:
Normally there is no need to wrap into a `main` function but since it's already the case for the other examples, let's just skip it. ;)